### PR TITLE
OPSIM-1173: : change basis function names, small bug fix

### PR DESCRIPTION
**Pull request by rhiannonlynne**
bug fix to AvoidFastRevisitsBasisFunction
update basis function names, along with backward compatibility
docstring extension

### CODE REVIEW

File: rubin_scheduler/scheduler/basis_functions/basis_functions.py [hunk 1]
```diff
@@ -1208,7 +1234,7 @@ def __init__(
         self.aways_available = aways_available
 
         self.survey_features = {}
-        self.survey_features["Last_observation"] = features.Last_observation()
+        self.survey_features["Last_observation"] = features.LastObservation()
```
Comment (yoachim) [inline rubin_scheduler/scheduler/basis_functions/basis_functions.py:1237]: Looks like `GoalStrictFilterBasisFunction` can get a dep warning or deleted too.

**Review by yoachim — APPROVED** [review]
I'd say just delete the basis functions that have been clearly languishing broken, or at least slap deprecation warnings more liberally. 

Checked and 5-year baseline sim matches one run on main.

File: rubin_scheduler/scheduler/basis_functions/basis_functions.py [hunk 2]
```diff
@@ -892,8 +894,8 @@ def __init__(self, nside=32, filtername1="r", filtername2="z", gap_min=40.0, gap
         self.gap_max = IntRounded(gap_max / 60.0 / 24.0)
 
         self.survey_features = {}
-        self.survey_features["last_obs_f1"] = features.Last_observed(filtername=filtername1, nside=nside)
-        self.survey_features["last_obs_f2"] = features.Last_observed(filtername=filtername2, nside=nside)
+        self.survey_features["last_obs_f1"] = features.LastObserved(filtername=filtername1, nside=nside)
+        self.survey_features["last_obs_f2"] = features.LastObserved(filtername=filtername2, nside=nside)
```
[the remark this answers is not in the document]
Comment (rhiannonlynne) [reply rubin_scheduler/scheduler/basis_functions/basis_functions.py:898]: I was really inclined to go straight to deleting once they started failing on instantiation.
I'll delete the one that won't even import; I'll start deleting the ones which don't get complaints, before the next tag.

File: rubin_scheduler/scheduler/basis_functions/basis_functions.py (hunk 1 above)
Comment (rhiannonlynne) [reply rubin_scheduler/scheduler/basis_functions/basis_functions.py:1237]: Hmm. It should be sending a warning  -- `send_unused_deprecation_warning(self.__class__.__name__)` is at the end of the init.

File: rubin_scheduler/scheduler/basis_functions/basis_functions.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (rhiannonlynne) [reply rubin_scheduler/scheduler/basis_functions/basis_functions.py:898]: (the one that won't instantiate easily is FootprintRollingBasisFunction .. it has a kwarg which is set to None, but None doesn't work .. I don't really want to go through and add the extra work to modify this one too,  if it was so clearly broken)

**Pull request merged**